waku/waku_api/rest/admin/handlers.nim

```diff
@@ -0,0 +1,119 @@
+    return err(RestApiResponse.badRequest("Wrong Content-Type, expected application/json"))
+
+  let reqBodyData = contentBody.get().data
+
+  let requestResult = decodeFromJsonBytes(T, reqBodyData)
+  if requestResult.isErr():
+    return err(RestApiResponse.badRequest("Invalid content body, could not decode. " &
+                                          $requestResult.error))
+
+  return ok(requestResult.get())
+
+proc tuplesToWakuPeers(peers: var WakuPeers, peersTup: seq[PeerProtocolTuple]) =
+  for peer in peersTup:
+    peers.add(peer.multiaddr, peer.protocol, peer.connected)
+
+
+proc installAdminV1GetPeersHandler(router: var RestRouter, node: WakuNode) =
+  router.api(MethodGet, ROUTE_ADMIN_V1_PEERS) do () -> RestApiResponse:
+    var peers:  WakuPeers = @[]
+
+    if not node.wakuRelay.isNil():
+      # Map managed peers to WakuPeers and add to return list
+      let relayPeers = node.peerManager
+                            .peerStore.peers(WakuRelayCodec)
+                            .mapIt((
+                                    multiaddr: constructMultiaddrStr(it),
+                                    protocol: WakuRelayCodec,
+                                    connected: it.connectedness == Connectedness.Connected)
+                                  )
+      tuplesToWakuPeers(peers, relayPeers)
+
+    if not node.wakuFilterLegacy.isNil():
+      # Map WakuFilter peers to WakuPeers and add to return list
+      let filterPeers = node.peerManager.peerStore.peers(WakuLegacyFilterCodec)
+          .mapIt((multiaddr: constructMultiaddrStr(it),
+                  protocol: WakuLegacyFilterCodec,
+                  connected: it.connectedness == Connectedness.Connected))
+      tuplesToWakuPeers(peers, filterPeers)
+
+    if not node.wakuStore.isNil():
+      # Map WakuStore peers to WakuPeers and add to return list
+      let storePeers = node.peerManager.peerStore
+                           .peers(WakuStoreCodec)
+                           .mapIt((multiaddr: constructMultiaddrStr(it),
+                                   protocol: WakuStoreCodec,
+                                   connected: it.connectedness == Connectedness.Connected))
+      tuplesToWakuPeers(peers, storePeers)
+
+    let resp = RestApiResponse.jsonResponse(peers, status=Http200)
+    if resp.isErr():
+      error "An error ocurred while building the json respose: ", error=resp.error
+      return RestApiResponse.internalServerError(fmt("An error ocurred while building the json respose: {resp.error}"))
+
+    return resp.get()
+
+proc installAdminV1PostPeersHandler(router: var RestRouter, node: WakuNode) =
+  router.api(MethodPost, ROUTE_ADMIN_V1_PEERS) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+
+    let decodedBody = decodeRequestBody[seq[string]](contentBody)
+
+    if decodedBody.isErr():
+      return RestApiResponse.badRequest(fmt("Failed to decode request: {decodedBody.error}"))
+
+    let peers: seq[string] = decodedBody.value()
```

Comment:
```suggestion
     let peers: seq[string] = decodeRequestBody[seq[string]](contentBody).isOkOr:
      return RestApiResponse.badRequest(fmt("Failed to decode request: {error}"))
```

Reply:
Cool, thanks! I like this more expressive way.